src/components/popups/FollowersPopup.vue

```diff
@@ -1,48 +1,35 @@
 <script setup lang="ts">
-import { ref, computed } from 'vue';
-import type { Profile } from '@/backend/profile';
-import { useProfilesStore } from '@/store/profiles';
+import { ref, computed, onBeforeMount } from 'vue';
 import { useRoute } from 'vue-router';
-import { useStore } from '@/store/session';
 import HorizontalProfilePreview from '@/components/HorizontalProfilePreview.vue';
 import CloseIcon from '@/components/icons/CloseIcon.vue';
+import { useProfilesStore } from '@/store/profiles';
+import { useStore } from '@/store/session';
+import { useConnectionsStore } from '@/store/connections';
 
 const profilesStore = useProfilesStore();
 const route = useRoute();
 const store = useStore();
 const emit = defineEmits([`close`]);
+const connections = useConnectionsStore();
 
 if (route.name !== `Home`) {
 	if (typeof route.params.id !== 'string') {
 		throw new Error('Invalid param type for id');
 	}
 }
 
-const authorID = route.name === `Home` ? store.$state.id : route.params.id;
+const authorID: string = route.name === `Home` ? store.$state.id : (route.params.id as string);
+profilesStore.fetchProfile(authorID);
+
 const profile = computed(() => profilesStore.getProfile(authorID as string));
+const followersList = ref<Set<string> | undefined>();
```

Comment:
Here, you can use computed property with a getter from store to fetch the followers for a profile, (similar to how we do for profile in line number 25)